sphinx_immaterial/content_tabs.py

```diff
@@ -123,23 +122,26 @@ def depart_tab_label(self, node):
 
 def visit_tab_set(self: HTMLTranslator, node: content_tab_set):
     # increment tab set counter
-    self.tab_set_count = getattr(self, "tab_set_count", 0) + 1
+    self.tab_set_count: int = getattr(self, "tab_set_count", 0) + 1  # type: ignore
 
     # configure tab set's div attributes
-    tab_set_identity = f"__tabbed_{self.tab_set_count}"
-    attributes = {"data-tabs": f"{self.tab_set_count}:{len(node.children)}"}
+    tab_set_identity = f"__tabbed_{self.tab_set_count}"  # type: ignore
+    attributes = {"data-tabs": f"{self.tab_set_count}:{len(node.children)}"}  # type: ignore
```

Comment:
If it knows that self.tab_set_count is always an `int`, then we shouldn't need the `type: ignore` here. Unless `len(nodes.children)` is causing confusion in mypy.

Reply:
The `type: ignore` was in regards to the attribute `tab_set_count`.  mypy can't handle adding properties to a class outside of its definition, I believe.

Reply:
`# type: ignore` just suppresses all type checking errors on that line.  So in this case it isn't about the type of `tab_set_identity`, but about the access to `self.tab_set_count`.

Reply:
I think we could narrow the scope of `# type: ignore` by moving the access of `self.tab_set_count` to a separate line.

Reply:
I'm working locally on this branch, and I found that using the following doesn't require the type ignore comments
```py
    tab_set_count: int = getattr(self, "tab_set_count", 0) + 1
    setattr(self, "tab_set_count", tab_set_count)

    # configure tab set's div attributes
    tab_set_identity = f"__tabbed_{tab_set_count}"
    attributes = {"data-tabs": f"{tab_set_count}:{len(node.children)}"}
```

---
I'm also going through the other changes on this branch... I'll push when ready.

Reply:
I went forward with the `Union[base_t, derived_t]` because the type **hinting** isn't just programmatic for me (it helps during code review).